src/utils/dom.ts

```diff
@@ -10,6 +14,12 @@ export function getAttributes(el: HTMLElement, blockList: string[]): Record<stri
     .reduce((acc, { name, value }) => ({ ...acc, [name]: value }), {});
 }
 
+export function getThemeName(el: HTMLElement): Theme {
+  return getComputedStyle(el)
```

Comment:
this causes reflow of an element, which may cause reflow to elements around it. If you're calling this as part of a render loop it may be good to stress test it and do some memory auditing

Reply:
Updated to use closest() and cross shadow.